File: .codex_review_extract/laravel-api/app/Models/Expense.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class Expense extends Model
{
    protected $table = 'pengeluaran';
    protected $primaryKey = 'id_pengeluaran';
    public $timestamps = false;

    protected $fillable = [
        'id_shift',
        'kategori',
        'nominal',
        'keterangan',
        'tanggal',
    ];

    protected $casts = [
        'nominal' => 'decimal:2',
        'tanggal' => 'datetime',
    ];

    public function shift(): BelongsTo
    {
        return $this->belongsTo(Shift::class, 'id_shift', 'id_shift');
    }
}

File: .codex_review_extract/flutter-pos/lib/data/models/expense.dart
import 'num_util.dart';
class Expense {
  final int idPengeluaran;
  final int idShift;
  final String kategori;
  final double nominal;
  final String? keterangan;
  final String tanggal;

  const Expense({
    required this.idPengeluaran,
    required this.idShift,
    required this.kategori,
    required this.nominal,
    this.keterangan,
    required this.tanggal,
  });

  factory Expense.fromJson(Map<String, dynamic> json) => Expense(
        idPengeluaran: json['id_pengeluaran'] as int,
        idShift: json['id_shift'] as int,
        kategori: json['kategori'] as String,
        nominal: parseDouble(json['nominal']),
        keterangan: json['keterangan'] as String?,
        tanggal: json['tanggal'] as String,
      );

  Map<String, dynamic> toJson() => {
        'id_pengeluaran': idPengeluaran,
        'id_shift': idShift,
        'kategori': kategori,
        'nominal': nominal,
        'keterangan': keterangan,
        'tanggal': tanggal,
      };
}
